config/validator.go

```diff
@@ -142,3 +142,36 @@ func RequestHeaderContentTypeValidator(req http.Request, requiredContentType str
 	}
 	return 0
 }
+
+
+// DomainsEntryLogicalValidator checks the correctness of "Domains" part of configuration file
+func (c *YamlConfig) DomainsEntryLogicalValidator() (valid bool, validationErrors map[string][]error) {
+	errList := make([]error, 0)
+	var domains []string
+	for _, regionConf := range c.Regions {
+		for _, regionDomain := range regionConf.Domains {
+			for _, alreadyDefinedDomain := range domains {
+				if regionDomain == alreadyDefinedDomain {
+					errList = append(
+						errList,
+						fmt.Errorf("Invalid domain %s! Domain already defined", regionDomain))
+				} else if strings.HasSuffix(regionDomain, "." + alreadyDefinedDomain) ||
+						  strings.HasSuffix(alreadyDefinedDomain, "." + regionDomain) {
+					errList = append(
+						errList,
+						fmt.Errorf("Invalid domain %s! Domain conflicts with %s", regionDomain, alreadyDefinedDomain))
```

Comment:
Maybe it does not follow some conventions, but what about \n at the end to make it more readable? (and the same with another message above).